code/model/modelread.cpp

```diff
@@ -3395,24 +3451,26 @@ void model_rot_sub_into_obj(vec3d * outpnt, vec3d *mpnt,polymodel *pm, int sub_m
 }
 
 
-// Given a rotating submodel, find the ship and world axes or rotatation.
-void model_get_rotating_submodel_axis(vec3d *model_axis, vec3d *world_axis, int modelnum, int submodel_num, object *obj)
+// Given a rotating submodel, find the local and world axes of rotatation.
+// If orient is NULL, "world" axis is local to the model's reference frame
+void model_get_rotating_submodel_axis(vec3d *model_axis, vec3d *world_axis, matrix *orient, int model_num, int model_instance_num, int submodel_num)
 {
-	polymodel *pm = model_get(modelnum);
-
+	polymodel *pm = model_get(model_num);
 	bsp_info *sm = &pm->submodel[submodel_num];
-	Assert(sm->movement_type == MOVEMENT_TYPE_ROT || sm->movement_type == MOVEMENT_TYPE_DUMB_ROTATE);
+	Assert(sm->movement_type == MOVEMENT_TYPE_ROT || sm->movement_type == MOVEMENT_TYPE_INTRINSIC_ROTATE);
 
 	if (sm->movement_axis == MOVEMENT_AXIS_X) {
 		vm_vec_make(model_axis, 1.0f, 0.0f, 0.0f);
 	} else if (sm->movement_axis == MOVEMENT_AXIS_Y) {
 		vm_vec_make(model_axis, 0.0f, 1.0f, 0.0f);
-	} else {
-		Assert(sm->movement_axis == MOVEMENT_AXIS_Z);
+	} else if (sm->movement_axis == MOVEMENT_AXIS_Z) {
 		vm_vec_make(model_axis, 0.0f, 0.0f, 1.0f);
+	} else {
+		// must be one of these axes
+		Int3();
```

Comment:
This should be replaced by an Assertion that includes a better description of the problem. `Int3()` simply crashes the process without any information about the cause.